client/src/pages/settings-panel/data/text-editor-settings.ts

```diff
@@ -36,6 +36,79 @@ export const textEditorSettings: Group = {
         },
       ],
     },
+    font: {
+      title: 'Font',
+      description: 'Customize font settings.',
+      configs: [
+        {
+          type: ConfigType.INPUT,
+          title: 'Font Family',
+          description: 'Controls the font family.',
+          default: DefaultTextEditorSettings.fontFamily as string,
+          key: 'fontFamily',
+        },
+        {
+          type: ConfigType.INPUT,
+          title: 'Font Size',
+          description: 'Controls the font size.',
+          default: DefaultTextEditorSettings.fontSize as number,
+          key: 'fontSize',
+        },
+        {
+          type: ConfigType.INPUT,
+          title: 'Line Height',
+          description: 'Controls the line height.',
+          default: DefaultTextEditorSettings.lineHeight as string,
+          key: 'lineHeight',
+        },
+        {
+          type: ConfigType.INPUT,
+          title: 'Font Weight',
+          description: 'Controls the font size.',
+          default: DefaultTextEditorSettings.fontWeight as number,
+          key: 'fontWeight',
+        },
+        {
+          type: ConfigType.CHECKBOX,
+          title: 'Font Ligature',
+          description: 'Controls whether font ligature is used',
+          default: DefaultTextEditorSettings.fontLigatures as boolean,
+          key: 'fontLigatures',
+        },
+      ],
+    },
+    customBehaviour: {
+      title: 'Custom Behaviour',
+      description: 'Change editor behaviour and themes.',
+      configs: [
+        {
+          type: ConfigType.SELECT,
+          title: 'Custom Bindings',
+          description: 'Controls the binding used by the editor',
+          default: DefaultTextEditorSettings.customBinding as string,
+          options: [
+            ['None', 'none'],
```

Comment:
As these values are used explicitly in the `BaseEditor` component, we should make them as enums. 

```typescript
enum Keybinding = {
  VIM = 'vim',
  EMACS = 'emacs',
  NONE = 'none'
}
```